### PR TITLE
feat(core): make ExternalsValidator optional for the FeatureServiceRegistry

### PR DESCRIPTION
**Pull request by KingHenne**
- [x] rebase and set base branch to `master` after #341 has been merged

### CODE REVIEW

File: packages/core/src/feature-service-registry.ts [hunk 1]
```diff
@@ -109,6 +109,20 @@ export interface FeatureServiceRegistryOptions {
    * registered.
    */
   readonly configs?: FeatureServiceConfigs;
+
+  /**
+   * When the {@link FeatureAppManager} is configured with a
```
Comment (clebert) [inline packages/core/src/feature-service-registry.ts]: If the

File: packages/core/src/feature-service-registry.ts [hunk 2]
```diff
@@ -109,6 +109,20 @@ export interface FeatureServiceRegistryOptions {
...
+   * {@link FeatureAppManager#moduleLoader}, to load Feature Apps from a remote
+   * location that also provide their own Feature Services, i.e. the Feature
+   * Services are included in a different bundle than the integrator bundle, it
+   * might make sense to validate external dependencies that are required by
+   * those Feature Services against the shared dependencies that are provided by
+   * the integrator. This makes it possible that an error is already thrown when
```
Comment (clebert) [inline packages/core/src/feature-service-registry.ts]: This makes it possible to detect this incompatibility when registering a Feature Service with incompatible external dependencies.

File: packages/core/src/feature-service-registry.ts [hunk 3]
```diff
@@ -109,6 +109,20 @@ export interface FeatureServiceRegistryOptions {
...
+   * registering a Feature Service with incompatible external dependencies. This
+   * gives the author early feedback as to whether a Feature Service is
```
Comment (clebert) [inline packages/core/src/feature-service-registry.ts]: -author

File: packages/core/src/feature-app-manager.ts [hunk 4]
```diff
@@ -70,8 +70,28 @@ export interface FeatureAppManagerLike {
...
+   * `@feature-hub/module-loader-commonjs` package).
+   */
   readonly moduleLoader?: ModuleLoader;
+
+  /**
+   * When using a {@link #moduleLoader}, it might make sense to validate
+   * external dependencies that are required by Feature Apps against the
+   * shared dependencies that are provided by the integrator. This makes it
+   * possible that an error is already thrown when creating a Feature App with
+   * incompatible external dependencies. This gives the author early feedback as
```
Comment (clebert) [inline packages/core/src/feature-app-manager.ts]: -author

File: packages/core/src/__tests__/feature-service-registry.test.ts [hunk 5]
```diff
@@ -339,65 +336,109 @@ describe('FeatureServiceRegistry', () => {
       );
     });
 
-    it('fails to register a Feature Service due to failing externals validation', () => {
-      const mockError = new Error('mockError');
-
-      mockExternalsValidator.validate = jest.fn(() => {
-        throw mockError;
+    describe('with an ExternalsValidator provided to the FeatureServiceRegistry', () => {
```
Comment (clebert) [inline packages/core/src/__tests__/feature-service-registry.test.ts]: see missing test case, in former PR

File: packages/core/src/feature-service-registry.ts (hunk 2 above)
Comment (KingHenne) [reply packages/core/src/feature-service-registry.ts]: I would prefer a similar sentence to the one we decided on for the `FeatureAppManagerOptions`:

> This makes it possible that an error is already thrown when registering a Feature Service with incompatible external dependencies, and thus enables early feedback as to whether a Feature Service is compatible with the integration environment.

File: packages/core/src/feature-service-registry.ts (hunk 3 above)
Comment (KingHenne) [reply packages/core/src/feature-service-registry.ts]: fixed

File: packages/core/src/feature-app-manager.ts (hunk 4 above)
Comment (KingHenne) [reply packages/core/src/feature-app-manager.ts]: fixed

File: packages/core/src/__tests__/feature-service-registry.test.ts (hunk 5 above)
Comment (KingHenne) [reply packages/core/src/__tests__/feature-service-registry.test.ts]: done

File: packages/core/src/feature-service-registry.ts (hunk 1 above)
Comment (KingHenne) [reply packages/core/src/feature-service-registry.ts]: fixed

**Pull request merged**